src/engine/cache/file-cache.js

```diff
@@ -54,18 +54,26 @@ class FileCache extends BaseCache {
     await createFolder(this.fileFolder)
     await createFolder(this.errorFolder)
 
-    const files = await fs.readdir(this.fileFolder)
-    if (files.length > 0) {
-      this.logger.debug(`${files.length} files in cache.`)
-    } else {
-      this.logger.debug('No files in cache.')
+    try {
+      const files = await fs.readdir(this.fileFolder)
+      if (files.length > 0) {
+        this.logger.debug(`${files.length} files in cache.`)
+      } else {
+        this.logger.debug('No files in cache.')
+      }
+    } catch (error) {
+      this.logger.error(error)
```

Comment:
a comment explaining why this can happen and why silently ignoring it is the right way to go would be more than helpful